server/backend/testspg/router/test_router_user_role_assignment.py

```diff
@@ -428,103 +423,312 @@ def test_user_role_forest_client_number_not_exist_bad_request(
+    will not be able to remove access with other forest client numbers like FOM_SUBMITTER_00001011
+    """
+    # create a user role assignment for a business bceid user within the same organization as the user COGNITO_USERNAME_BCEID
+    # with abstract role FOM_SUBMITTER and forest client number 00001018
+    user_role_xref_id = create_test_user_role_assignment(
+        test_client_fixture,
+        fom_dev_access_admin_token,
+        TEST_USER_ROLE_ASSIGNMENT_FOM_DEV_ABSTRACT_BCEID,
+    )
+
+    # create a token for business bceid user COGNITO_USERNAME_BCEID with no app admin role
+    # we already grant this user the delegated admin privilege for role FOM_SUBMITTER_00001018
+    token = jwt_utils.create_jwt_token(
+        test_rsa_key, [], jwt_utils.COGNITO_USERNAME_BCEID
+    )
+    response = test_client_fixture.delete(
+        f"{endPoint}/{user_role_xref_id}",
+        headers=jwt_utils.headers(token),
+    )
+    # delete is successful
+    assert response.status_code == HTTPStatus.NO_CONTENT
+
+    # create a user role assignment for a business bceid user within the same organization as the user COGNITO_USERNAME_BCEID
+    # with abstract role FOM_SUBMITTER and forest client number 00001011
+    user_role_xref_id = create_test_user_role_assignment(
+        test_client_fixture,
+        fom_dev_access_admin_token,
+        {**TEST_USER_ROLE_ASSIGNMENT_FOM_DEV_ABSTRACT_BCEID, "forest_client_number": "00001011"},
+    )
+    response = test_client_fixture.delete(
+        f"{endPoint}/{user_role_xref_id}",
+        headers=jwt_utils.headers(token),
+    )
+    assert response.status_code == HTTPStatus.FORBIDDEN
+    assert response.json() is not None
+    data = response.json()
+    # business bceid user has no privilege to delete role with forest client number 00001011
+    assert data["detail"]["code"] == ERROR_PERMISSION_REQUIRED
+
+
+def test_deleter_user_role_assignment_bceid_cannot_delete_idir_access(
+    test_client_fixture: starlette.testclient.TestClient,
+    fom_dev_access_admin_token,
+    test_rsa_key,
+):
+    """
+    test business bceid user cannnot delete idir user access
+    """
+    # create a user role assignment for IDIR user
+    user_role_xref_id = create_test_user_role_assignment(
+        test_client_fixture,
+        fom_dev_access_admin_token,
+        TEST_USER_ROLE_ASSIGNMENT_FOM_DEV_CONCRETE,
+    )
+
+    # create a token for business bceid user COGNITO_USERNAME_BCEID with FOM_DEV_ADMIN role
+    token = jwt_utils.create_jwt_token(
+        test_rsa_key, [FOM_DEV_ADMIN_ROLE], jwt_utils.COGNITO_USERNAME_BCEID
+    )
+    response = test_client_fixture.delete(
+        f"{endPoint}/{user_role_xref_id}",
+        headers=jwt_utils.headers(token),
+    )
+    assert response.status_code == HTTPStatus.FORBIDDEN
+    assert response.json() is not None
+    data = response.json()
+    # business bceid user cannot delete idir user access
+    assert data["detail"]["code"] == ERROR_PERMISSION_REQUIRED
```

Comment:
Hi Catherine, I think maybe in this case, we should check the "**description**"(`data["details"]["description"]`) and not the "**code**" since several other guards return also "**ERROR_PERMISSION_REQUIRED**".

Reply:
yeah I agree, updated these test cases and added the check for description, thanks!